File: HolaMundoJava/app/src/main/java/com/example/holamundojava/CalculadoraBasicaActivity.java
package com.example.holamundojava;


import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.example.holamundojava.backend.OperacionesBasicas;

public class CalculadoraBasicaActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Button buttonUno, buttonDos, buttonTres, buttonCuatro, buttonCinco, buttonSeis, buttonSiete, buttonOcho, buttonNueve, buttonCero, buttonSuma, buttonResta, buttonMultiplicacion, buttonDivision, buttonIgual, buttonPunto, buttonBorrar;
        EditText editTextNumbers;
        Button buttonSalir;

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculadora_basica);

        buttonUno = findViewById(R.id.buttonUno);
        buttonDos = findViewById(R.id.buttonDos);
        buttonTres = findViewById(R.id.buttonTres);
        buttonCuatro = findViewById(R.id.buttonCuatro);
        buttonCinco = findViewById(R.id.buttonCinco);
        buttonSeis = findViewById(R.id.buttonSeis);
        buttonSiete = findViewById(R.id.buttonSiete);
        buttonOcho = findViewById(R.id.buttonOcho);
        buttonNueve = findViewById(R.id.buttonNueve);
        buttonCero = findViewById(R.id.buttonCero);
        buttonSuma = findViewById(R.id.buttonSumar);
        buttonResta = findViewById(R.id.buttonRestar);
        buttonMultiplicacion = findViewById(R.id.buttonMultiplicar);
        buttonDivision = findViewById(R.id.buttonDividir);
        buttonIgual = findViewById(R.id.buttonIgual);
        buttonBorrar = findViewById(R.id.buttonLimpiar);
        editTextNumbers = findViewById(R.id.editTextNumbers);
        buttonSalir = findViewById(R.id.buttonSalirCalculadoraBasica);
        buttonUno.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "1");
            }
        });
        buttonDos.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "2");
            }
        });
        buttonTres.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "3");
            }
        });
        buttonCuatro.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "4");
            }
        });
        buttonCinco.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "5");
            }
        });
        buttonSeis.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "6");
            }
        });
        buttonSiete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "7");
            }
        });
        buttonOcho.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "8");
            }
        });
        buttonNueve.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "9");
            }
        });
        buttonCero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "0");
            }
        });
        buttonSuma.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "+");
            }
        });
        buttonResta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "-");
            }
        });
        buttonMultiplicacion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "*");
            }
        });
        buttonDivision.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText(editTextNumbers.getText() + "/");
            }
        });
        buttonIgual.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                OperacionesBasicas operacionesBasicas = new OperacionesBasicas();
                String cadena = editTextNumbers.getText().toString();
                String[] elementos = cadena.split("[-+*/]");

                if (elementos.length == 2) {
                    int num1 = Integer.parseInt(elementos[0]);
                    int num2 = Integer.parseInt(elementos[1]);

                    if (cadena.contains("+")) {
                        editTextNumbers.setText(String.valueOf(operacionesBasicas.suma(num1, num2)));
                    } else if (cadena.contains("-")) {
                        editTextNumbers.setText(String.valueOf(operacionesBasicas.resta(num1, num2)));
                    } else if (cadena.contains("*")) {
                        editTextNumbers.setText(String.valueOf(operacionesBasicas.multiplicacion(num1, num2)));
                    } else if (cadena.contains("/")) {
                        if (num2 != 0) {
                            editTextNumbers.setText(String.valueOf(operacionesBasicas.division(num1, num2)));
                        } else {
                            editTextNumbers.setText("No se puede dividir por cero");
                        }
                    }
                }
            }
        });

        buttonBorrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editTextNumbers.setText("");
            }
        });
        buttonSalir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }
}

File: HolaMundoKotlin/app/src/main/java/com/example/holamundokotlin/CalculadoraBasicaActivity.kt
package com.example.holamundokotlin

import android.annotation.SuppressLint
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.example.holamundokotlin.backend.OperacionesBasicas

class CalculadoraBasicaActivity : AppCompatActivity() {
    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculadora_basica)

        val buttonUno: Button = findViewById(R.id.buttonUno)
        val buttonDos: Button = findViewById(R.id.buttonDos)
        val buttonTres: Button = findViewById(R.id.buttonTres)
        val buttonCuatro: Button = findViewById(R.id.buttonCuatro)
        val buttonCinco: Button = findViewById(R.id.buttonCinco)
        val buttonSeis: Button = findViewById(R.id.buttonSeis)
        val buttonSiete: Button = findViewById(R.id.buttonSiete)
        val buttonOcho: Button = findViewById(R.id.buttonOcho)
        val buttonNueve: Button = findViewById(R.id.buttonNueve)
        val buttonCero: Button = findViewById(R.id.buttonCero)
        val buttonSuma: Button = findViewById(R.id.buttonSumar)
        val buttonResta: Button = findViewById(R.id.buttonRestar)
        val buttonMultiplicacion: Button = findViewById(R.id.buttonMultiplicar)
        val buttonDivision: Button = findViewById(R.id.buttonDividir)
        val buttonIgual: Button = findViewById(R.id.buttonIgual)
        val buttonBorrar: Button = findViewById(R.id.buttonLimpiar)
        val editTextNumbers: EditText = findViewById(R.id.editTextNumbers)
        val buttonSalir: Button = findViewById(R.id.buttonSalirCalculadoraBasica)
        buttonUno.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "1"
            )
        }
        buttonDos.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "2"
            )
        }
        buttonTres.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "3"
            )
        }
        buttonCuatro.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "4"
            )
        }
        buttonCinco.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "5"
            )
        }
        buttonSeis.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "6"
            )
        }
        buttonSiete.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "7"
            )
        }
        buttonOcho.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "8"
            )
        }
        buttonNueve.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "9"
            )
        }
        buttonCero.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "0"
            )
        }
        buttonSuma.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "+"
            )
        }
        buttonResta.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "-"
            )
        }
        buttonMultiplicacion.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "*"
            )
        }
        buttonDivision.setOnClickListener {
            editTextNumbers.setText(
                editTextNumbers.getText().toString() + "/"
            )
        }
        buttonIgual.setOnClickListener {
            val operacionesBasicas = OperacionesBasicas()
            val cadena = editTextNumbers.text.toString()
            val elementos = cadena.split("[-+*/]".toRegex())

            if (elementos.size == 2) {
                val num1 = elementos[0].toInt()
                val num2 = elementos[1].toInt()
                if(cadena.contains('+')){
                    editTextNumbers.setText(operacionesBasicas.suma(num1, num2).toString())
                } else if(cadena.contains('-')){
                    editTextNumbers.setText(operacionesBasicas.resta(num1, num2).toString())
                } else if(cadena.contains('*')){
                    editTextNumbers.setText(operacionesBasicas.multiplicacion(num1, num2).toString())
                } else if(cadena.contains('/')){
                    if (num2 != 0) {
                        editTextNumbers.setText(operacionesBasicas.division(num1, num2))
                    } else {
                        editTextNumbers.setText("No se puede dividir por cero")
                    }
                }
            }
        }

        buttonBorrar.setOnClickListener { editTextNumbers.setText("") }
        buttonSalir.setOnClickListener { finish() }
    }
}